include/geom/node_elem.h

```diff
@@ -106,7 +106,7 @@ class NodeElem : public Elem
    * This should never be important for NodeElems.
    */
   virtual dof_id_type key (const unsigned int) const libmesh_override
-  { return 0; }
+  { libmesh_not_implemented(); return 0; }
```

Comment:
This should be libmesh_error(), not libmesh_not_implemented(), right?  libmesh_not_implemented() is for corner cases where the call is sensible but we just haven't written the code, whereas I suspect calling key() on a NodeElem will always be an error..